### PR TITLE
Check technology eligibility #84

### PR DESCRIPTION
**Pull request by ami-day**
Added a function which checks the following:

- Are the technology ontology labels all eligible based on SCEA criteria and the intended functionality of the script?
Raise an assertion error if 1 or more technology types are not eligible.
Expected error message: "AssertionError: 1 or more technology types are not eligible. Please remove ineligible technologies and their linked samples and try again."

- Is there more than 1 unique technology ontology label in the dataset?
If more than 1 technology type, raise an assertion error.
Expected error message:  "AssertionError: Only 1 technology type is allowed per SCEA E-HCAD id. Please split the dataset by the technology type and run them separately."

Tested the changes on test cases 01-06 listed here: https://docs.google.com/spreadsheets/d/1i6nG6I-Sc9VS2Z_6Hz0ErRidxU8Ui6iCFBnB0CzncMA/edit#gid=0
The expected error messages are raised as intended.

### CODE REVIEW

File: hca-to-scea/hca2scea.py [hunk 1]
```diff
@@ -552,8 +552,7 @@ def main():
     xlsx_dict = multitab_excel_to_single_txt.remove_unused_protocols(xlsx_dict)
 
     '''Check whether multiple library preparation protocol technology types or 10X versions were
-    used. If so, split xlsx_dict into a list of dicts separated by the technology type. Then,
-    create idf and sdrf files for each of the dicts.'''
+    used. Raise an assertion error if so. The user will need to split their dataset by technology type.'''
```
Comment (amnonkhen) [inline hca-to-scea/hca2scea.py]: Comments get stale very quickly.
Your code should be packed into functions in the appropriate level of detail that are named properly and convey your intentions.

File: hca-to-scea/hca2scea.py [hunk 2]
```diff
@@ -585,55 +585,40 @@ def main():
...
-        if pooled_samples:
-            print("The hca-to-scea tool does not support pooled donors or pooled samples."
-                "The dataset should be curated manually.")
-            sys.exit()
+    ''' Check if samples are pooled '''
+    pooled_samples = check_experimental_design.check_for_pooled_samples(xlsx_dict)
+    if pooled_samples:
+        print("The hca-to-scea tool does not support pooled donors or pooled samples."
+            "The dataset should be curated manually.")
+        sys.exit()
```
Comment (amnonkhen) [inline hca-to-scea/hca2scea.py]: Using sys.exit makes it diffiult to integrate your code into something else at a later stage, e.g. running it from the UI.
A better way is to raise an exception, and catch it higher up in the call chain.

**Review by amnonkhen — CHANGES_REQUESTED** [review]

File: hca-to-scea/hca2scea.py (hunk 2 above)
Comment (ami-day) [reply hca-to-scea/hca2scea.py]: This code snippet has actually been removed from the code in a separate branch which checks for pooled samples. In that branch, an exception is raised and it occurs as part of the experimental design checks before the main script functionality.

File: hca-to-scea/hca2scea.py (hunk 1 above)
Comment (ami-day) [reply hca-to-scea/hca2scea.py]: "Your code should be packed into functions in the appropriate level of detail that are named properly" - this is what I have been aiming for, but I have added comments in the script on top of this. For example, the function name to check technology eligibility is: check_technology_eligibility(xlsx_dict,technology_dict). Is it best to remove all the comments above functions in the main script then?

**Review by amnonkhen — APPROVED** [review]

**Pull request merged**